evm_loader/program/src/executor/state.rs

```diff
@@ -506,6 +529,15 @@ impl<'a, B: AccountStorage> Database for ExecutorState<'a, B> {
                 }
             }
         }
+        // put offset to cache
+        self.data.cache.borrow_mut().actions_offset = self.data.actions.len();
+        for (addr, acc) in &accounts {
+            self.data
+                .cache
+                .borrow_mut()
+                .accounts
+                .insert(*addr, acc.clone()); // put accounts to cache
```

Comment:
Please avoid cloning an account here.

Reply:
Also, you only need to update the cache with accounts that have been changed in emulated actions.